ocaml/xapi/sm.ml

```diff
@@ -132,9 +132,11 @@ let sr_update dconf driver sr =
   let call = Sm_exec.make_call ~sr_ref:sr dconf "sr_update" [] in
   Sm_exec.parse_unit (Sm_exec.exec_xmlrpc (driver_filename driver) call)
 
-let vdi_create dconf driver sr sm_config vdi_type size name_label
+let vdi_create ?traceparent dconf driver sr sm_config vdi_type size name_label
     name_description metadata_of_pool is_a_snapshot snapshot_time snapshot_of
     read_only =
+  Tracing.with_tracing ?parent:(Option.some traceparent) ~name:"SM.vdi_create"
```

Comment:
Please try to see if this works:
```suggestion
  Tracing.with_tracing ~parent:traceparent ~name:"SM.vdi_create"
```

Reply:
I will try it.

Reply:
This commit https://github.com/xapi-project/xen-api/pull/5293/commits/81310de0ec3c3af9a8e00ed239b782c65556ea1b should work now. Also, after talking with @mg12, I've made a second commit which use `with_dbg` instead of `with_tracing`.